src/scala/com/github/johnynek/bazel_deps/MakeDeps.scala

```diff
@@ -199,17 +201,18 @@ object MakeDeps {
     }
   }
 
-  private def executeGenerate(model: Model, projectRoot: File, workspacePath: IO.Path, workspaceContents: String, targets: List[Target], formatter: Writer.BuildFileFormatter): Unit = {
+  private def executeGenerate(model: Model, projectRoot: File, workspacePath: IO.Path, targetFileOpt: Option[IO.Path], enable3rdPartyInRepo: Boolean, workspaceContents: String, targets: List[Target], formatter: Writer.BuildFileFormatter): Unit = {
     // Build up the IO operations that need to run. Till now,
     // nothing was written
     val buildFileName = model.getOptions.getBuildFileName
     val io = for {
       originalBuildFile <- IO.readUtf8(workspacePath.sibling(buildFileName))
-      _ <- IO.recursiveRmF(IO.Path(model.getOptions.getThirdPartyDirectory.parts))
+      // If the 3rdparty directory is empty we shouldn't wipe out the current working directory.
+      _ <- if(enable3rdPartyInRepo && model.getOptions.getThirdPartyDirectory.parts.nonEmpty) IO.recursiveRmF(IO.Path(model.getOptions.getThirdPartyDirectory.parts), false) else IO.const(1)
```

Comment:
I think `IO.const(0)` might be a better since I think it is count of the number of deleted files.

Reply:
👍